cmd/scollector/collectors/snmp_ciscobgp.go

```diff
@@ -0,0 +1,190 @@
+package collectors
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	"bosun.org/_third_party/github.com/mjibson/snmp"
+	"bosun.org/cmd/scollector/conf"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+)
+
+const (
+	state               = ".1.3.6.1.4.1.9.9.187.1.2.5.1.3.1.4"
+	adminStatus         = ".1.3.6.1.4.1.9.9.187.1.2.5.1.4.1.4"
+	localAS             = ".1.3.6.1.4.1.9.9.187.1.2.5.1.8.1.4"
+	localIdentifier     = ".1.3.6.1.4.1.9.9.187.1.2.5.1.9.1.4"
+	remoteAS            = ".1.3.6.1.4.1.9.9.187.1.2.5.1.11.1.4"
+	remoteIdentifier    = ".1.3.6.1.4.1.9.9.187.1.2.5.1.12.1.4"
+	inUpdates           = ".1.3.6.1.4.1.9.9.187.1.2.5.1.13.1.4"
+	outUpdates          = ".1.3.6.1.4.1.9.9.187.1.2.5.1.14.1.4"
+	inTotalMessages     = ".1.3.6.1.4.1.9.9.187.1.2.5.1.15.1.4"
+	outTotalMessages    = ".1.3.6.1.4.1.9.9.187.1.2.5.1.16.1.4"
+	inUpdateElapsedTime = ".1.3.6.1.4.1.9.9.187.1.2.5.1.27.1.4"
+	acceptedPrefixes    = ".1.3.6.1.4.1.9.9.187.1.2.8.1.1.1.4"
+	deniedPrefixes      = ".1.3.6.1.4.1.9.9.187.1.2.8.1.2.1.4"
+	advertisedPrefixes  = ".1.3.6.1.4.1.9.9.187.1.2.8.1.6.1.4"
+	suppressedPrefixes  = ".1.3.6.1.4.1.9.9.187.1.2.8.1.7.1.4"
+	withdrawnPrefixes   = ".1.3.6.1.4.1.9.9.187.1.2.8.1.8.1.4"
+)
+
+func SNMPCiscoBGP(cfg conf.SNMP) {
+	collectors = append(collectors, &IntervalCollector{
+		F: func() (opentsdb.MultiDataPoint, error) {
+			return c_snmp_ciscobgp(cfg.Community, cfg.Host)
+		},
+		Interval: time.Second * 30,
+		name:     fmt.Sprintf("snmp-ciscobgp-%s", cfg.Host),
+	})
+}
+
+func c_snmp_ciscobgp(community, host string) (opentsdb.MultiDataPoint, error) {
+	// Tag: local_as
+	localASesRaw, err := snmp_ip_tree(host, community, localAS)
+	if err != nil {
+		return nil, err
+	}
+	localASes := make(map[string]string, len(localASesRaw))
+	for k, v := range localASesRaw {
+		s := strconv.FormatInt(v.(int64), 10)
+		if err != nil {
+			return nil, err
+		}
+		localASes[k] = s
+	}
+
+	// Tag: local_id
+	localIdentifiersRaw, err := snmp_ip_tree(host, community, localIdentifier)
+	if err != nil {
+		return nil, err
+	}
+	localIdentifiers := make(map[string]string, len(localIdentifiersRaw))
+	for k, v := range localIdentifiersRaw {
+		localIdentifiers[k] = snmp_combine_ip_uint8(v.([]uint8))
```

Comment:
Can you clarify? A value from that OID should always be []uint8 - should I switch things around so we won't get a panic on bad data?